### ISSUE TITLE
Token metadata timeouts

### ISSUE THREAD

**Issue by klemenfn** [blockscout/blockscout#5384]
*Describe your issue here.*

### Environment

* Blockscout Version/branch: 4.1.2-beta

We see a lot of these errors in the log:
```
00:11:21.922 [debug] failed to fetch token instance metadata for {"0xe...", 4484}: {:error, :timeout}
```

Digging through the code, these seem to be timeouts when fetching assets from ipfs. Similar errors appear on ipfs gateways  which rate limit the requests due to fetcher retries.

It seems that HTTPoison client uses default timeout of 5 seconds in [apps/explorer/lib/explorer/token/instance_metadata_retriever.ex](https://github.com/blockscout/blockscout/blob/ce99db3956b64f7db569677fbc3da4302398fd93/apps/explorer/lib/explorer/token/instance_metadata_retriever.ex#L207). Would that be the correct location to increase it? IPFS can be incredibly slow so I am thinking to just bump this to 30s or more to see if it solves the problem. If set too high this could cause too many open connections but I can try to find a balanced value.

More importantly, we see see that in rare instances, Token Transfer is listed correctly (/token-transfers) but Token is not shown in the inventory (/tokens). This means blockscout correctly indexed the token transfer transaction but not the inventory. **If token instance metadata fetch fails, would that cause token not appearing in the inventory?**

Missing inventory seems to sync eventually after some time (hours or days).

**Comment by vbaranov** [comment]
@klemenfn yes, you can increase the default 5_000 timeout in the line, you posted

https://github.com/blockscout/blockscout/blob/ce99db3956b64f7db569677fbc3da4302398fd93/apps/explorer/lib/explorer/token/instance_metadata_retriever.ex#L207

```
HTTPoison.get(url, [], [timeout: 50_000, recv_timeout: 50_000])
```

I think we will increase it in the master branch as well. I agree with you, that it will be useful for IPFS links.

**Comment by klemenfn** [comment]
Does the failure to fetch the asset also prevent the token to be listed in the inventory? I couldn't deduce that logic from the code on my own.

**Comment by vbaranov** [comment]
> Does the failure to fetch the asset also prevent the token to be listed in the address token list? I couldn't deduce that logic from the code on my own.

No, these are independent processes. Also, vrom v5.3.0 of Blockscout backend, it is possible to provide the URL of your own IPFS gateway via `IPFS_GATEWAY_URL` - this may speed up token instance metadata fetch from IPFS.

Moreover, the latest version of Blockscout ([v6](https://github.com/blockscout/blockscout/releases/tag/v6.5.0-beta)) contains split of the queues of the token instances to realtime/retry/sanitize, prioritizing of errors to re-fetch, which speed-up the metadata import process in overall.
